### PR TITLE
Implement ActiveModel::Attributes on context

### PR DESCRIPTION
**Pull request by aaronmallen**
## Description

Implement the `ActiveModel::Attributes` API on `ActiveInteractor::Context::Base`

## Information

- [x] Contains Documentation
- [x] Contains Tests
- [ ] Contains Breaking Changes

## Related Issues

- see #137

### Added

- `ActiveInteractor::Context::Attributes#attribute?`
- `ActiveInteractor::Context::Attributes#has_attribute?`
- `ActiveInteractor::Context::Base#attribute_missing`
- `ActiveInteractor::Context::Base#attribute_names`
- `ActiveInteractor::Context::Base#method_missing`
- `ActiveInteractor::Context::Base#respond_to?`
- `ActiveInteractor::Context::Base#respond_to_without_attributes?`
- `ActiveInteractor::Context::Base.attribute`
- `ActiveInteractor::Context::Base.attribute_missing`
- `ActiveInteractor::Context::Base.attribute_names`
- `ActiveInteractor::Context::Base.method_missing`
- `ActiveInteractor::Context::Base.respond_to?`
- `ActiveInteractor::Context::Base.respond_to_without_attributes?`
- `ActiveInteractor::Interactor::Context#context_attribute_missing`
- `ActiveInteractor::Interactor::Context#context_attribute_names`
- `ActiveInteractor::Interactor::Context#context_respond_to_without_attributes?`
- `ActiveInteractor::Interactor::Context::ClassMethods#context_attribute`
- `ActiveInteractor::Interactor::Context::ClassMethods#context_attribute_missing`
- `ActiveInteractor::Interactor::Context::ClassMethods#context_attribute_names`
- `ActiveInteractor::Interactor::Context::ClassMethods#context_respond_to_without_attributes?`

### CODE REVIEW

**Comment by aaronmallen** [comment]
Hoping @dark-panda will give this patch a look.  I tested this on a local example-app with `shoulda-matchers` and it seems to work perfectly.

File: spec/active_interactor/context/base_spec.rb [hunk 1]
```diff
@@ -5,32 +5,70 @@
 RSpec.describe ActiveInteractor::Context::Base do
   after(:each) { described_class.instance_variable_set('@__attributes', []) }
```
Comment (dark-panda) [inline spec/active_interactor/context/base_spec.rb]: Looks like some left over code here.

Comment (aaronmallen) [reply spec/active_interactor/context/base_spec.rb]: Good catch!

**Comment by dark-panda** [comment]
Looks good to me overall, and works with my specs in my test application.

**Pull request merged**